lib/diagnostic.js

```diff
@@ -6,29 +6,73 @@
 // splits it into an array of normalized words
 // (uppercase strings without punctuation)
 // and returns that array.
-const normalizeWords = function (/* */) {
-  /* your response here */
+let normalizedWords = [];
+let getUniqueWords = [];
+
+const normalizeWords = function ( text )
+{
+  let words = text.split(/\s+/);
+  for (let i = 0; i < words.length; i++)
+  {
+    normalizedWords[i] = words[i].replace( /\W+/, '' ).toUpperCase();
+  }
+  return normalizedWords;
 };
 
 // 2. Write a function that takes a string argument
 // and returns an array of unique normalized words.
-const uniqueWords = function (/* */) {
-  /* your response here */
+const uniqueWords = function ( text )
+{
+  let uniqueWordsAsKeys = {};
+  for (let i = 0; i < normalizedWords.length; i++)
+  {
+      uniqueWordsAsKeys[normalizedWords[i]] = true;
+  }
+
+  let keys = [];
+  let k = 0;
+
+
+  for( keys[k++] in uniqueWordsAsKeys ); // jshint ignore:line
+
+  getUniqueWords = keys;
+
+  return getUniqueWords;
+
 };
 
 // 3. Write a function that takes a string argument and returns the count of
 // words in a string.
 // Provide the *option* to count unique words instead of total words, by passing
 // a second argument.
 
-const wordCount = function (/* */) {
-  /* your response here */
+const wordCount = function ( text, flag )
+{
+  let lengthString = 0;
+  if( flag === true )
+  {
+    lengthString = uniqueWords( text ).length;
+  }
+  else
+  {
+    lengthString = normalizeWords( text ).length;
+  }
+  return lengthString;
 };
```

Comment:
this is written correctly, but it's not working because your uniqueWords function doesn't return the right thing